src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeService.java

```diff
@@ -132,7 +132,18 @@ public void saveEvent(UUID envelopeId, EventType eventType) {
     }
 
     @Transactional(readOnly = true)
-    public List<Envelope> getEnvelopes(String blobName, String containerName, LocalDate date) {
-        return envelopeRepository.findEnvelopes(blobName, containerName, date);
+    public List<Tuple2<Envelope, List<EnvelopeEvent>>> getEnvelopes(
```

Comment:
Can we perhaps create a special class to represent envelope info?

Reply:
We already have `EnvelopeInfo` class. the result will be mapped to `EnvelopeInfo` class in the controller

Reply:
fair enough

Reply:
Do you think it would be better to replace this by a "join" on envelopes and envelope_events and have a mapper class?

Reply:
It would be one query instead of hundreds, so a good idea.

Reply:
Either a join with a mapper or two queries - one retrieving envelopes and the other retrieving all events for those envelopes.

Reply:
cool, I'll update the pr

Reply:
I've updated to get the data in two queries.